File: src/types/PLayerState.ts

// tutaj muszę lepiej dopasowac bo ma byc 1:1 z backendem
export interface PlayerState 
{
    isPlaying: boolean
    progressInMs: number,
    repeatState: string,
    shuffleStateOn: boolean,
    track: {
        trackName: string,
        artistNames: string[],
        spotifyTrackUrl: string,
        durationInMin: number
    }
}

File: src/main/java/com/domain/model/PlayerState/PlayerState.java
package com.domain.model.PlayerState;

import com.domain.model.Track.TrackModel;
import com.domain.model.Device.Device;

public record PlayerState(TrackModel track,
                          Device device,
                          String repeatState,
                          int timestamp,
                          int progressInMs,
                          boolean isPlaying,
                          boolean shuffleStateOn
                          ) {}
